src/test/java/org/folio/rest/api/InstanceBulkTest.java

```diff
@@ -0,0 +1,400 @@
+      assertThat(moons.containsKey(idObject.getString("id")), is(true));
+    }
+  }
+
+  private void setInstanceSequence(long sequenceNumber) {
+    final Vertx vertx = StorageTestSuite.getVertx();
+    final PostgresClient postgresClient =
+        PostgresClient.getInstance(vertx, TENANT_ID);
+    final CompletableFuture<Void> sequenceSet = new CompletableFuture<>();
+
+    vertx.runOnContext(v -> {
+      postgresClient.selectSingle("select setval('hrid_instances_seq',"
+          + sequenceNumber + ",FALSE)", r -> {
+            if (r.succeeded()) {
+              sequenceSet.complete(null);
+            } else {
+              sequenceSet.completeExceptionally(r.cause());
+            }
+          });
+    });
+
+    try {
+      sequenceSet.get(2, SECONDS);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void createInstance(JsonObject instanceToCreate)
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+
+    client.post(instancesStorageUrl(""), instanceToCreate,
+      TENANT_ID, json(createCompleted));
+
+    Response response = createCompleted.get(2, SECONDS);
+
+    assertThat(String.format("Create instance failed: %s", response.getBody()),
+      response.getStatusCode(), is(201));
+  }
+
+  private void createManyMoons(Map<String, JsonObject> instancesToCreate)
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    for (String id : instancesToCreate.keySet()) {
+      createInstance(instancesToCreate.get(id));
+    }
+  }
+
+  private Map<String, JsonObject> manyMoons(int total) {
+    return manyMoons(total, InstanceBulkIdsGetField.ID);
+  }
+
+  private Map<String, JsonObject> manyMoons(int total,
+      InstanceBulkIdsGetField field) {
+    Map<String, JsonObject> moons = new HashMap<String, JsonObject>();
+
+    List<String[]> languages = Arrays.asList(new String[][] {
+      new String[] {"eng"},
+      new String[] {"ger"},
+      new String[] {"por", "ger"}
+    });
+
+    for (int i = 0; i < total; i++) {
+      UUID uuid = UUID.randomUUID();
+      String hrid = String.format("in%011d", i + 1);
+
+      String id = "";
+      if (field.compareTo(InstanceBulkIdsGetField.ID) == 0) {
+        id = uuid.toString();
+      }
+
+      JsonArray identifiers = new JsonArray();
+      identifiers.add(identifier(UUID_ISBN, "9781473619777"));
+      JsonArray contributors = new JsonArray();
+      contributors.add(contributor(UUID_PERSONAL_NAME, "Chambers, Becky"));
+      JsonArray tags = new JsonArray();
+      tags.add("test-tag");
+
+      JsonArray langs = new JsonArray();
+      int l = 0;
+      if (i % 3 == 0) {
+        l = 2;
+      }
+      else if (i % 2 == 0) {
+        l = 1;
+      }
+      for (int j = 0; j < languages.get(l).length; j++) {
```

Comment:
Is it important for the test to have this randomization for language? Just think if we can simplify it a bit.

Reply:
It is not random, it is just using every threes, every evens, and every odds and rotating between them.

I originally wrote this when I was expecting to use the `lang` url parameter.
I wanted a simple and easy way to have some arbitrary but predictable language generator.

I was uncertain if I would still need to test languages via the query parameter after considering [MODINVSTOR-198](https://issues.folio.org/browse/MODINVSTOR-198).

Should I remove this entirely and the language tests for now?

Reply:
I have not seen any logic related to this attribute, I think it is fine to remove/simplify this test.